### ISSUE TITLE
Improper Typecheck / Ambiguous Name

### ISSUE THREAD

**Issue by Ratismal** [middric/add-one#2]
This fails when trying to add one to a float.
```js
const addOne = require('add-one');
const number = 1.1;
console.log(addOne(number));
// Expected: 2.1
// Actual: throw Error("Not an integer.")
```
The cause of this is one of two:
1. The typecheck to ensure that a number is provided is done improperly (see: `is-integer`)
2. This library is only intended to add one to integers, in which case the name is ambiguous and should be revised (ex. `add-one-to-integer`, alternative libraries can be created for `add-one-to-float`)

Please address this, as it is blocking my workflow. Thank you!

**Comment by natanbc** [comment]
Yes, this is a very needed improvement, our backend is crashing due to not being able to increase a float counter.
